src/runtime/util.cpp

```diff
@@ -46,25 +46,25 @@ void parseSlice(BoxedSlice* slice, int size, i64* out_start, i64* out_stop, i64*
         }
     }
 
-    if (isSubclass(start->cls, int_cls)) {
-        istart = static_cast<BoxedInt*>(start)->n;
-        if (istart < 0)
-            istart = size + istart;
-    } else {
+    if (start->cls == none_cls || (start->cls != int_cls && !isSubclass(start->cls, int_cls))) {
```

Comment:
I'd like to spend some time with them in the future (once verifying that they're still needed after the `isSubclass` work), but definitely not something needed for this change.